### PR TITLE
Added VSINPU ExecutionProvider for onnxruntime

### PR DESCRIPTION
**Pull request by chenfeiyue-cfy**
Added VSINPU files in onnxruntime
Modify onnxruntime source code for model input/output name matching

Type: New Feature

### CODE REVIEW

**Review by sunshinemyson — APPROVED** [review]

File: onnxruntime/core/framework/ort_value_name_idx_map.h [hunk 1]
```diff
@@ -39,7 +39,19 @@ class OrtValueNameIdxMap {
     auto it = map_.find(name);
 #endif
     if (it == map_.end()) {
-      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Could not find OrtValue with name '", name, "'");
```
Comment (thezha) [inline onnxruntime/core/framework/ort_value_name_idx_map.h:42]: What is this change? Why is this change necessary?

File: onnxruntime/core/framework/session_state.cc [hunk 2]
```diff
@@ -903,7 +903,19 @@ common::Status SessionState::GetInputNodeInfo(const std::string& input_name,
                                               InlinedVector<NodeInfo>& node_info_vec) const {
   auto entry = input_names_to_nodeinfo_mapping_.find(input_name);
   if (entry == input_names_to_nodeinfo_mapping_.cend()) {
-    return Status(ONNXRUNTIME, FAIL, "Failed to find input name in the mapping: " + input_name);
```
Comment (thezha) [inline onnxruntime/core/framework/session_state.cc:906]: What is this change? Why is this change necessary?

Comment (chenfeiyue-cfy) [reply onnxruntime/core/framework/session_state.cc:906]: Hi kainan, the purpose of this code is to use string matching when the map key(Type:String) cannot be found, because some of the model's test data provided by onnxruntime repo itself cannot be found due to **inconsistent naming**. However, if you think this hack method is not formal, I will restore the source code.
